src/textures/TextTexture.mjs

```diff
@@ -428,7 +429,9 @@ export default class TextTexture extends Texture {
 
         return function(cb) {
             const canvas = this.stage.platform.getDrawingCanvas();
-            const renderer = new TextTextureRenderer(this.stage, canvas, args);
+            const renderer = Utils.isSpark ?
+                new TextTextureRendererSpark(this.stage, canvas, args) :
```

Comment:
Since spark platform implementation code is in Lightning-Spark, wouldn't it make more sense to move TextTextureRendererSpark there aswell?

Reply:
I added the file the the platform area in src/platforms/spark because it depends TextTextureRenderer.  Is this not the best place to put it (given its dependency on Lightning's TextTextureRenderer)?  If you feel that the best spot is to move from the platforms area of Lightning into the Lightning-Spark repo then I will do it.  Please let me know.

Reply:
I see. Ideal solution would be for lightning to be platform agnostic (or just for web) and have forks of Lightning with different implementations (node or spark) OR have all platform support inside upstream lightning. From what I've seen, we currently have somewhat inbetween not very elegant solution so let me think a little bit about best solution here.

Reply:
Ok, I think it's best to have TextTextureRendererSpark and copy of TextTextureRenderer in Lightning-Spark.

Reply:
Moving TextTextureRendererSpark and a copy of TextTextureRenderer out of Lightning and into Lightning-Spark repository became pretty complex because TextTextureRenderer also had a dependency on StageUtils and Utils and bringing them over as well didn't seem like the best option.  Also, making an update to TextTextureRenderer will then require an update to the same class in the Lightning-Spark repo.

To make things simpler, I removed the TextTextureRendererSpark class and added a platform.drawText method the Lightning-Spark repo.  This should keep the changes small and clean while also achieving your requirement of having all the Spark specific code inside its own repository.  Please review and let me know if any additional changes are needed.  Thanks!